service/migration_manager.cc

```diff
@@ -1090,6 +1091,7 @@ future<schema_ptr> get_schema_definition(table_schema_version v, netw::messaging
                 // It should be fine since if we tried to register a view for which
                 // we don't know the base table, our registry is broken.
                 schema_ptr base_schema = db.find_schema(s->view_info()->base_id());
+                s = db::schema_tables::maybe_fix_legacy_secondary_index_mv_schema(get_local_storage_proxy().local_db(), view_ptr(s), base_schema, db::schema_tables::preserve_version::yes);
```

Comment:
I think you can't just unconditionally take the return value from here, you need to check if it's not null. Here's an excerpt from the implementation:
https://github.com/scylladb/scylla/blob/dbd0b9a3efe8eaaa436586bfa9be050a601915ad/db/schema_tables.cc#L3119-L3122
So in the (quite likely I think) case when the schema is ok, the line below would dereference a null pointer.

Reply:
Ohh you are right. Will fix it right away.